### PR TITLE
Api provides functionality for visitors to see comments

### PR DESCRIPTION
**Pull request by LucasKnudsen**
[**PT-Story:**  ](https://www.pivotaltracker.com/story/show/178309529)
### User Story 
``` 
As an API
In order for users to see other people's opinions
I need to be able to serve the client with comments
``` 
## Changes proposed in this pull request: 
* Adds Comment model
* Adds associations with User and Article
* Adds custom JSON format for the show serializer
* Adds comments count to index serializer
* Adds total comments count to statistics controller

## What I have learned working on this feature: 
* Some more experience creating object formats with loops :)

### CODE REVIEW

File: spec/models/article_spec.rb [hunk 1]
```diff
@@ -44,8 +44,9 @@
     }
   end
 
-  describe 'Relationship between article and user' do
+  describe 'Relationships' do
```
Comment (1CIM) [inline spec/models/article_spec.rb:47]: <3

File: spec/requests/api/editor/editor_can_see_all_articles_spec.rb [hunk 2]
```diff
@@ -3,11 +3,12 @@
   let!(:journalist) { create(:user, role: 'journalist') }
   let!(:journalist2) { create(:user, role: 'journalist') }
   let(:auth_headers) { editor.create_new_auth_token }
-
+  
   describe 'successfully' do
     let!(:article1) { create(:article, title: 'First Article', user_id: journalist.id) }
     let!(:article2) { create(:article, title: 'Second Article', user_id: journalist.id) }
     let!(:article3) { create(:article, title: 'Third Article', user_id: journalist2.id) }
+    let!(:comment) { create(:comment, article_id: article3.id) }
```
Comment (1CIM) [inline spec/requests/api/editor/editor_can_see_all_articles_spec.rb:11]: might wanna rename them to "something_num.id" instead as You Know Who said earlier today? ;)

File: spec/requests/api/editor/editor_can_see_company_statistics_spec.rb [hunk 3]
```diff
@@ -3,9 +3,10 @@
   let!(:editor_headers) { editor.create_new_auth_token }
   let!(:journalist) { create(:user, role: 'journalist') }
   let!(:journalist_headers) { journalist.create_new_auth_token }
-  let!(:published_articles1) { create(:article, created_at: Time.zone.now) }
-  let!(:published_articles2) { create(:article, created_at: Time.zone.now - 100_000)  }
-  let!(:published_articles3) { create(:article, created_at: Time.zone.now - 200_000)  }
+  let!(:published_article1) { create(:article, created_at: Time.zone.now) }
+  let!(:published_article2) { create(:article, created_at: Time.zone.now - 100_000)  }
+  let!(:published_article3) { create(:article, created_at: Time.zone.now - 200_000)  }
```
Comment (1CIM) [inline spec/requests/api/editor/editor_can_see_company_statistics_spec.rb:8]: what dose 100_000 stuff do?

**Review by 1CIM — APPROVED** [review]

File: spec/requests/api/editor/editor_can_see_company_statistics_spec.rb (hunk 3 above)
Comment (LucasKnudsen) [reply spec/requests/api/editor/editor_can_see_company_statistics_spec.rb:8]: It's just a separator apparently, done by Solar :p

File: spec/requests/api/editor/editor_can_see_all_articles_spec.rb (hunk 2 above)
Comment (LucasKnudsen) [reply spec/requests/api/editor/editor_can_see_all_articles_spec.rb:11]: Iz true. Let's sort that in our big refactor session :3

Comment (1CIM) [reply spec/requests/api/editor/editor_can_see_all_articles_spec.rb:11]: Mmmmm sounds tasty with that BRS 🤤

**Review by 1CIM — APPROVED** [review]

**Review by christerforsgren91 — APPROVED** [review]

**Pull request merged**